lib/elixir/test/elixir/enum_test.exs

```diff
@@ -600,9 +600,31 @@ defmodule EnumTest do
     assert Enum.min_max_by(%{}, fn x -> String.length(x) end, fn -> {:no_min, :no_max} end) ==
              {:no_min, :no_max}
 
+    assert Enum.min_max_by(["aaa", "a", "aa"], fn x -> String.length(x) end, &>/2) == {"aaa", "a"}
+
     assert_runs_enumeration_only_once(&Enum.min_max_by(&1, fn x -> x end, fn -> nil end))
   end
 
+  test "min_max_by/4" do
+    users = [%{id: 1, date: ~D[2019-01-01]}, %{id: 2, date: ~D[2020-01-01]}]
+
+    assert Enum.min_max_by(users, & &1.date, Date) ==
+             {%{id: 1, date: ~D[2019-01-01]}, %{id: 2, date: ~D[2020-01-01]}}
```

Comment:
Note: min/max etc. also check that it correctly handles equal values, but those functions use a different sorter, and `Enum.min_max_by` has never had this behaviour as far as I could see.